tiledb/api/c_api/array_schema/CMakeLists.txt

```
@@ -0,0 +1,42 @@
+#
+# tiledb/api/c_api/array_schema/CMakeLists.txt
+#
+# The MIT License
+#
+# Copyright (c) 2024 TileDB, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+include(common NO_POLICY_SCOPE)
+include(object_library)
+
+list(APPEND SOURCES
+  array_schema_api.cc
+)
+gather_sources(${SOURCES})
+
+commence(object_library capi_array_schema_stub)
+  this_target_sources(${SOURCES})
+  this_target_link_libraries(export)
+  this_target_object_libraries(array array_schema)
```

Where does it depend on `array`?

---

More importantly, this handle really needs to not depend on `array` to avoid potential problems with link reference cycles.

---

This should be `array_directory`, not `array`. It's an oversight from before the `array` OL PR was merged (thanks for catching...). I'll make the change. 
To be clear, the dependency on `array_directory` is in `tiledb_array_schema_load`.

---

Here's what this line needs to look like:
```suggestion
  this_target_object_libraries(array_schema)
```
It needs no dependency on `array`, and no dependency on `array_directory`. If we have either of those present here, we're going to end up with cyclic dependencies.

This is a symptom of something that's outside of the CMake specification. If `capi_array_schema_stub` doesn't link when `array` and `array_directory` are both absent, then there's a substantive dependency that needs to be identified and redressed.

---

I'd forgotten I moved `tiled_array_schema_load` into `tiledb/api/c_api/array/array_api.cc` to remove this dependency. I was thinking that API was still in the schema section, requiring the dependency. Thanks for clarifying, and sorry for the confusion. I'm making this update now.